server/views/pages/prisonperson/prisonPersonMigrationFailures.njk

```diff
@@ -0,0 +1,89 @@
+        <div class="govuk-grid-column-full">
+          <p>All records that are on the main dead letter queue are displayed. These may include records from other migration runs.
+            If a record that failed at the point of the migration was subsequently fixed (or discarded) then it will no longer be shown on this page.
+          </p>
+
+        {% if failures.messages %}
+        {% set rows = [] %}
+        {% for message in failures.messages %}
+          {% set cellHtml %}
+            <div>
+              <div class="govuk-grid-row govuk-!-margin-bottom-2">
+                <div class="govuk-grid-column-three-quarters">
+                  <span class="govuk-!-font-weight-bold govuk-!-padding-right-2">ID</span>{{message.messageId}}
+                </div>
+              </div>
+              <div class="govuk-grid-row govuk-!-margin-bottom-2">
+                <div class="govuk-grid-column-three-quarters">
+                  <span class="govuk-!-font-weight-bold govuk-!-padding-right-2">Migration ID</span>{{message.body.context.migrationId}}
+                </div>
+                <div class="govuk-grid-column-one-quarter pull-right">
+                  <span class="govuk-!-font-weight-bold govuk-!-padding-right-2">Type</span>{{message.body.type}}
+                </div>
+              </div>
+              <div class="govuk-grid-row govuk-!-margin-bottom-2">
+                <div class="govuk-grid-column-full">
+                  Get more details from <a href="{{message.applicationInsightsLink}}" target="_blank">Application Insights</a>
+                </div>
+              </div>
+              <div class="govuk-grid-row govuk-!-margin-bottom-2">
+                <div class="govuk-grid-column-full">
+                  {% if message.body.type %}
+                    <span class="govuk-!-font-weight-bold govuk-!-padding-right-2">Event ID:</span>{{message.body.context.body.eventId}}
+                  {% endif %}
+                </div>
+              </div>
+              <div class="govuk-grid-row govuk-!-margin-bottom-2">
+                <div class="govuk-grid-column-full">
+                {{ govukDetails({
+                  summaryText: "Full message",
+                  text: message | json(['applicationInsightsLink'])
+                }) }}                
+                </div>
+              </div>
+            </div>
+          {% endset %}
+
+          {% set rows = (rows.push([
+            { html: cellHtml }
+            ]), rows) %}
+          {% endfor %}
+
+          {% set bob %}
+            Failures displayed {{failures.messagesReturnedCount}} of {{failures.messagesFoundCount}}
+          {% endset %}
+
+          {{ govukTable({
+            caption: bob,
+            captionClasses: "govuk-table__caption--m",            
+            rows: rows
+          }) }}
+        {% else %}
+          <p data-qa=no-failures class="govuk-body">There no failures still waiting to be resolved</p>
```

Comment:
readability - should probably be there ARE no failures

Reply:
I'll change the other 8 pages that say this while I'm at it - and then think about why we are duplicating so much code.